packages/block-library/src/separator/block.json

```diff
@@ -8,16 +8,21 @@
 	"keywords": [ "horizontal-line", "hr", "divider" ],
 	"textdomain": "default",
 	"attributes": {
-		"color": {
-			"type": "string"
-		},
-		"customColor": {
+		"opacity": {
```

Comment:
Is it worth setting a default value here? (I guess that would be `alpha-channel`?)

Reply:
I tried that, but if you set a default then no attribute is serialised in the block content, so you then can't tell the difference between a new block with the default `alpha-channel` and a v1 block that has no attributes, eg.
```
<!-- wp:separator -->
<hr class="wp-block-separator"/>
<!-- /wp:separator -->
```

Reply:
Thanks @glendaviesnz, I think with the way the current PR is written, that markup would still get migrated, because it doesn't have the `has-alpha-channel-opacity` class name. So with the following changes, that markup will get treated as the v1 version of the block. Here's the diff I'm using:


<details>
<summary>Diff exploring setting a default</summary>

```diff
diff --git a/packages/block-library/src/separator/block.json b/packages/block-library/src/separator/block.json
index 672abc411e..97556aef1e 100644
--- a/packages/block-library/src/separator/block.json
+++ b/packages/block-library/src/separator/block.json
@@ -9,7 +9,8 @@
 	"textdomain": "default",
 	"attributes": {
 		"opacity": {
-			"type": "string"
+			"type": "string",
+			"default": "alpha-channel"
 		}
 	},
 	"supports": {
diff --git a/packages/block-library/src/separator/deprecated.js b/packages/block-library/src/separator/deprecated.js
index 8003e1ef42..011402dfa9 100644
--- a/packages/block-library/src/separator/deprecated.js
+++ b/packages/block-library/src/separator/deprecated.js
@@ -41,9 +41,6 @@ const v1 = {
 
 		return <hr { ...useBlockProps.save( { className, style } ) } />;
 	},
-	isEligible( { opacity } ) {
-		return ! opacity;
-	},
 	migrate( attributes ) {
 		const { color, customColor } = attributes;
 		return {
diff --git a/packages/block-library/src/separator/edit.js b/packages/block-library/src/separator/edit.js
index ce32585739..39dae46139 100644
--- a/packages/block-library/src/separator/edit.js
+++ b/packages/block-library/src/separator/edit.js
@@ -19,17 +19,6 @@ export default function SeparatorEdit( { attributes, setAttributes } ) {
 	const { backgroundColor, style, opacity } = attributes;
 	const color = useColorProps( attributes );
 
-	// V1 blocks expect to have a default opacity of 0.4, so on transformation
-	// get an opacity attribute set to css in order to assign a class name
-	// to provide this opactiy. Once the user changes the color this is removed
-	// and opacity can then be set via the color block support alpha channel.
-
-	useEffect( () => {
-		if ( ! opacity ) {
-			setAttributes( { opacity: 'alpha-channel' } );
-		}
-	}, [ opacity ] );
-
 	const previousClassName = usePrevious( color.className );
 	useEffect( () => {
 		if (

```
</details>

If I copy and paste the following in the code editor view:

```
<!-- wp:separator -->
<hr class="wp-block-separator"/>
<!-- /wp:separator -->
```

That then becomes the following:

```
<!-- wp:separator {"opacity":"css"} -->
<hr class="wp-block-separator has-css-opacity"/>
<!-- /wp:separator -->
```

Whereas a newly inserted separator is added as the following:

```
<!-- wp:separator -->
<hr class="wp-block-separator has-alpha-channel-opacity"/>
<!-- /wp:separator -->
```

So long as we're always adding that additional classname, it should work? But I wasn't sure if that was the intention there, either 🤔

Again, apologies if I'm missing something obvious here. Thanks for indulging the questions!